### ISSUE TITLE
"Accessed" date update

### ISSUE THREAD

**Issue by ftorto** [keepassxreboot/keepassxc#2005]
## Expected Behavior
When I copy the password or when I use autofill (*from keepassxc, not from browser with plugin*), I expect the `accessed` date of the entry to be updated to **now**.

## Current Behavior

The access date seems to be equal to the last modified date

## Possible Solution

- Precise the usage of access date if it's not related to my expected behaviour
- Add a new checkbox option in settings to activate the update upon autofill/copy password to clipboard actions

Operating system: Ubuntu 16.04
CPU architecture: x86_64
Kernel: 4.13.0-41

**Comment by droidmonkey** [comment]
The problem with "accessed" is that every time you interact with the database it would "dirty" it and require a save. If autosave is on, then it would save everytime you did anything. This is not a good behavior, especially if you are using a robust key derivation like Argon2.

My opinion is to get rid of accessed all together.

**Comment by ftorto** [comment]
I'm not against the idea to get rid of it. My objective was to track the *unused websites*. 
Another way to do this could be to set a default expiration date to X months and check very old expired passwords.
I can't be confident with "created date" because I just imported to keepassxc from a CSV format which didn't had this field.

**Comment by TheZ3ro** [comment]
> My opinion is to get rid of accessed all together.

We can't really do this since the accessed property is defined in the KDBX standard.

Using access date to track unused entry seems a great idea to me, but like droidmonkey said, this property will be updated every time and so the kdbx will.

Sincerely, I see the accessed attributes as an attribute tracked when an entry is opened (open the edit interface but no edit is made).

We should look at what other clients do with it.

**Comment by droidmonkey** [comment]
By get rid of it, I meant in the GUI only.

**Comment by CodingSpiderFox** [comment]
I'd like to have this feature working, not getting rid of it in the GUI.
It's a good measure to check after you let someone use your PC while you forgot to lock keepass whether the person did not steal a credential.

**Comment by evilSwanson** [comment]
I agree that if there's no easy way to fix this it should be removed from the GUI because it does nothing and is misleading.

**Comment by jbond00747** [comment]
I understand not wanting to change this value on every access and thus need to re-write the file. Would it be possible to set an option along the lines of "Update last accessed value if it's older than <X> (Hours|Days|etc.)" While this wouldn't show you the absolute latest access, it would provide a good balance between figuring out which accounts aren't being used over a long time and minimizing disk writes. (Conceptually this is similar to the relatime option for filesystems on Linux. (https://blog.confirm.ch/mount-options-atime-vs-relatime/))

**Comment by CarnageMarkus** [comment]
Would it make difference if accessed field was updated only when password is retrieved from entry ? So it wouldn't update every time entries are retrieved in say search search, or by other clients (browser) but really only when it was used to 'login' ?

**Comment by droidmonkey** [comment]
The database shouldn't be modified for merely reading it or using the data from it. Since accessed is a part of an entry, and thus creates entry history, it cannot be deferred until later like some of our other non-data changes. Overall this is being removed from view.

Do note that KeePass2 doesn't show or use accessed even though they defined it in the kdbx.
